src/app/pages/psbt-request/components/psbt-decoded-request/psbt-decoded-request-views/build-psbt-input-for-ui.ts

```diff
@@ -0,0 +1,46 @@
+import { BitcoinNetworkModes } from '@shared/constants';
+import { getAddressFromOutScript } from '@shared/crypto/bitcoin/bitcoin.utils';
+import { isDefined } from '@shared/utils';
+
+import { OrdApiInscriptionTxOutput } from '@app/query/bitcoin/ordinals/ordinals-aware-utxo.query';
+
+import { PsbtInput, PsbtInputForUi } from './use-psbt-decoded-request';
+
+function getPsbtInputValue(input: PsbtInput, unsignedUtxo?: OrdApiInscriptionTxOutput) {
+  if (isDefined(input.witnessUtxo)) return Number(input.witnessUtxo.amount);
+  if (isDefined(input.nonWitnessUtxo) && isDefined(unsignedUtxo)) return Number(unsignedUtxo.value);
+  return 0;
+}
+
+function getPsbtInputAddress(
+  input: PsbtInput,
+  network: BitcoinNetworkModes,
+  unsignedUtxo?: OrdApiInscriptionTxOutput
+) {
+  if (isDefined(input.witnessUtxo))
+    return getAddressFromOutScript(input.witnessUtxo.script, network);
+  if (isDefined(input.nonWitnessUtxo) && isDefined(unsignedUtxo)) return unsignedUtxo.address ?? '';
+  return '';
+}
+
+interface BuildPsbtInputForUiArgs {
+  network: BitcoinNetworkModes;
+  psbtInputs: PsbtInput[];
+  unsignedUtxos: (OrdApiInscriptionTxOutput | undefined)[];
+}
+export function buildPsbtInputsForUi({
+  network,
+  psbtInputs,
+  unsignedUtxos,
+}: BuildPsbtInputForUiArgs): PsbtInputForUi[] {
+  return psbtInputs.map((input, i) => {
+    const utxoAddress = getPsbtInputAddress(input, network, unsignedUtxos[i]);
+    const utxoValue = getPsbtInputValue(input, unsignedUtxos[i]);
+    return {
+      ...input,
+      address: utxoAddress,
+      value: utxoValue,
+      unsignedUtxo: unsignedUtxos[i],
+    };
+  });
```

Comment:
Not sure I follow what's happening here. How does the index of the psbtInput relate to the unsignedUtxo?

Reply:
There are unsigned inputs and inputs/outputs containing the witness/non-witness data on the psbt, both have to be used in dep on what is passed in the psbt. The unsigned input index is the same as the input index on the psbt.

Reply:
Fwiw, I think trying to handle how an app passes us a psbt and rendering it in the UI is error prone ...like trying to handle nonces. But, I have been trying to throw the red flag on it for awhile now ...which is why I put a comment on the PR that we should have something in place for when the logic doesn't appear to work and force the advanced view only (or something).